# Annotations databases

AnnotSV requires different data sources for the annotation and the ranking of SV. In order to provide a ready to start installation of AnnotSV, each annotation source (that do not require a commercial license) is automatically downloaded during the installation.

The aim of each of these sources is fully explained in the README file.

Annotation can be performed using either the GRCh37 or GRCh38 build of the human genome (user defined). Some of the annotations are linked to the gene name and thus provided independently of the genome build.

<br />

# Overview of the different types of annotation:

## a) Gene annotations
- ### [RefSeq](https://www.ncbi.nlm.nih.gov/refseq/), [ENSEMBL](https://www.ensembl.org/index.html)

## b) Regulatory Elements annotations
- ### [miRTargetLink 2.0](https://ccb-compute.cs.uni-saarland.de/mirtargetlink2), [GeneHancer](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5467550/), [EnhancerAtlas](http://www.enhanceratlas.org/), [RefSeq](https://www.ncbi.nlm.nih.gov/refseq/), [ENSEMBL](https://www.ensembl.org/index.html)

## c) Gene-based annotations
- ### [GenCC](https://thegencc.org/), [OMIM](https://www.omim.org/), [ACMG](https://www.acmg.net/), [ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/), [ClinGen](https://clinicalgenome.org/), [gnomAD](https://gnomad.broadinstitute.org/), [ExAC](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5210650/), [DDD](https://www.deciphergenomics.org/), [Exomiser](https://github.com/exomiser/Exomiser)

## d) Known pathogenic genes or genomic regions annotation
- ### [ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/), [ClinGen](https://clinicalgenome.org/), [dbVar](https://www.ncbi.nlm.nih.gov/dbvar/), [OMIM morbid](https://www.omim.org/)

## e) Known pathogenic SNV/indel annotations
- ### [ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/)

## f) Known benign genes or genomic regions annotation
- ### [gnomAD](https://gnomad.broadinstitute.org/), [ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/), [ClinGen](https://clinicalgenome.org/), [DGV](http://dgv.tcag.ca/dgv/app/home), [DDD](https://www.deciphergenomics.org/), [1000g](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4690620/), [Ira M. Hall’s lab , Children’s Mercy Research Institute](https://www.childrensmercy.org/childrens-mercy-research-institute/)

## g) Breakpoints annotations
- ### [GC content, Repeat, Segmental duplication, ENCODE blacklist, GAP, Cytoband](https://genome.ucsc.edu/)

## h) TAD boundaries annotations

<br />

## i) COSMIC annotations (not distributed)

<br />
